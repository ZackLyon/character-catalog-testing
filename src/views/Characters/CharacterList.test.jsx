import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import CharacterList from './CharacterList.jsx';

const mockServer = setupServer(
  rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
    return res(
      ctx.json({
        info: { pages: 1 },
        results: [
          {
            id: 1,
            image: 'something',
            name: 'Rick',
            species: 'human',
            status: 'alive',
          },
          {
            id: 2,
            image: 'something',
            name: 'Morty',
            species: 'human',
            status: 'alive',
          },
          {
            id: 3,
            image: 'something',
            name: 'PickleRick',
            species: 'pickle',
            status: 'human again',
          },
        ],
      })
    );
  })
);

beforeAll(() => mockServer.listen());

afterAll(() => mockServer.close());

it('should display a loading message, then display a requested character', async () => {
  const container = render(
    <MemoryRouter initialEntries={['/characters']}>
      <Route path="/characters">
        <CharacterList />
      </Route>
    </MemoryRouter>
  );

  screen.getByAltText(/loading/i);
  await screen.findAllByText(/Rick/gi);
  await screen.findByText(/Morty/gi);
  await screen.findByText(/PickleRick/gi);
  expect(container).toMatchSnapshot();
});
