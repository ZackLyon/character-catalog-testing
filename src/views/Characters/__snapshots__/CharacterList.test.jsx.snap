// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a loading message, then display a requested character 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Characters
         
      </h1>
      <nav
        class="container"
      >
        <button
          class="button"
          disabled=""
        >
          ⏪
        </button>
        <button
          class="button"
          disabled=""
        >
          ◀️
        </button>
        <span
          class="pageCount"
        >
          1
           / 
          1
           
        </span>
        <button
          class="button"
          disabled=""
        >
          ▶️
        </button>
        <button
          class="button"
          disabled=""
        >
          ⏩
        </button>
      </nav>
      <ul
        aria-label="characters"
        class="list"
      >
        <li
          class="item"
        >
          <a
            href="/characters/1"
          >
            <figure>
              <img
                alt="Rick"
                class="image"
                src="something"
              />
              <figcaption
                class="linkText"
              >
                Rick
              </figcaption>
            </figure>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/characters/2"
          >
            <figure>
              <img
                alt="Morty"
                class="image"
                src="something"
              />
              <figcaption
                class="linkText"
              >
                Morty
              </figcaption>
            </figure>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/characters/3"
          >
            <figure>
              <img
                alt="PickleRick"
                class="image"
                src="something"
              />
              <figcaption
                class="linkText"
              >
                PickleRick
              </figcaption>
            </figure>
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1>
      Characters
       
    </h1>
    <nav
      class="container"
    >
      <button
        class="button"
        disabled=""
      >
        ⏪
      </button>
      <button
        class="button"
        disabled=""
      >
        ◀️
      </button>
      <span
        class="pageCount"
      >
        1
         / 
        1
         
      </span>
      <button
        class="button"
        disabled=""
      >
        ▶️
      </button>
      <button
        class="button"
        disabled=""
      >
        ⏩
      </button>
    </nav>
    <ul
      aria-label="characters"
      class="list"
    >
      <li
        class="item"
      >
        <a
          href="/characters/1"
        >
          <figure>
            <img
              alt="Rick"
              class="image"
              src="something"
            />
            <figcaption
              class="linkText"
            >
              Rick
            </figcaption>
          </figure>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          href="/characters/2"
        >
          <figure>
            <img
              alt="Morty"
              class="image"
              src="something"
            />
            <figcaption
              class="linkText"
            >
              Morty
            </figcaption>
          </figure>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          href="/characters/3"
        >
          <figure>
            <img
              alt="PickleRick"
              class="image"
              src="something"
            />
            <figcaption
              class="linkText"
            >
              PickleRick
            </figcaption>
          </figure>
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
